frontend/pkg/frontend/node_pool.go

```diff
@@ -42,7 +42,7 @@ func (f *Frontend) CreateOrUpdateNodePool(writer http.ResponseWriter, request *h
 		return
 	}
 
-	nodePoolResourceID, err := ResourceIDFromContext(ctx)
+	resourceID, err := ResourceIDFromContext(ctx)
```

Comment:
nit: feel free to dismiss - but I think I personally prefer the old names (`nodePoolResourceID` and `nodePoolDoc`) because there are so many resources at play here (clusters, CSnodePools, HCPNodePools, etc)

Reply:
Yeah I take your point.  For months I've been trying to inch toward unifying the cluster and node pool request handlers but so far I've only managed to do so for GET requests.  I tend to make the logic (and variable names) as uniform as possible between the two but maybe that's at the expense of clarity.

Generally in this code base -- and this is just a vocabulary I've come up with -- "resource IDs" refer to Azure resources and "internal IDs" refer to Cluster Service resources.  The [InternalID](https://github.com/Azure/ARO-HCP/blob/main/internal/ocm/internalid.go) type is just the OCM API path (`/api/clusters_mgmt/v1/cluster/...`) with some convenience methods around it.